### PR TITLE
Added authType to loginWithBrowser metric

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

1) Added refreshCredential metric on successful refresh
2) Added authType to loginWithBrowser metric
3) Add signInPageOpened metrics

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/resources/telemetryOverride.json [hunk 1]
```diff
@@ -308,6 +308,15 @@
                 "reloaded"
             ],
             "description": "Toolkit run state."
+        },
+        {
+            "name": "authType",
```
Comment (rli) [inline plugins/core/jetbrains-community/resources/telemetryOverride.json:313]: why not in common?

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt [hunk 2]
```diff
@@ -399,6 +402,13 @@ class SsoAccessTokenProvider(
 
             saveAccessToken(token)
 
+            sendRefreshCredentialsMetric(
+                currentToken,
+                result = Result.Succeeded,
+                reason = "",
+                reasonDesc = ""
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:409]: should be null
```suggestion
                reason = "",
                reasonDesc = ""
```

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 3]
```diff
@@ -401,4 +412,13 @@ abstract class LoginBrowser(
...
     }
+
+    fun isPKCEorDAG(region: String = "us-east-1"): AuthType {
+        val isCommercialRegion = !region.startsWith("us-gov") && !region.startsWith("us-iso") && !region.startsWith("cn")
+        if (!Registry.`is`("aws.dev.useDAG") && isCommercialRegion) {
+            return AuthType.PKCE
+        } else {
+            return AuthType.DeviceCode
+        }
+    }
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:423]: make `SsoAccessTokenProvider` use this logic too

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 4]
```diff
@@ -401,4 +412,13 @@ abstract class LoginBrowser(
             """.trimIndent()
         }
     }
+
+    fun isPKCEorDAG(region: String = "us-east-1"): AuthType {
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:416]: `is` implies boolean, but return is not boolean

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt (hunk 3 above)
Comment (manodnyab) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:423]: SsoAccessTokenProvider is not using this metric. SsoLoginCallBackProvider implements the copyCode functionality which is restricted to device code hence didnt add this login there

Comment (rli) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:423]: sorry i meant just do that we have 1 util doing the PKCE or DAG decision and not needing to remember to do it in two places

File: plugins/core/jetbrains-community/resources/telemetryOverride.json (hunk 1 above)
Comment (manodnyab) [reply plugins/core/jetbrains-community/resources/telemetryOverride.json:313]: We have diverged from the common implementation of this metric in JB. There is a backlog item to address this divergence

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt (hunk 3 above)
Comment (manodnyab) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt]: Oh moved the function to a util file

**Review by rli — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**